server/enterprise/message_export/global_relay_export/global_relay_export.go

```diff
@@ -72,57 +72,75 @@ type Message struct {
 	PreviewsPost   string
 }
 
-func GlobalRelayExport(rctx request.CTX, p shared.ExportParams) (results shared.RunExportResults, err error) {
+func GlobalRelayExport(rctx request.CTX, p shared.ExportParams) (shared.RunExportResults, error) {
+	results := shared.RunExportResults{}
+	var attachmentsRemovedPostIDs []string
+	allExports := make(map[string][]*ChannelExport)
+
 	tmpFile, err := os.CreateTemp("", "")
 	if err != nil {
 		return results, fmt.Errorf("unable to open the temporary export file: %w", err)
 	}
 	defer file.DeleteTemp(rctx.Logger(), tmpFile)
 
-	var attachmentsRemovedPostIDs []string
-	allExports := make(map[string][]*ChannelExport)
-
 	zipFile := zip.NewWriter(tmpFile)
 
+	// postAuthorsByChannel is a map so that we don't store duplicate authors
 	postAuthorsByChannel := make(map[string]map[string]shared.ChannelMember)
 	metadata := shared.Metadata{
-		Channels:         map[string]*shared.MetadataChannel{},
+		Channels:         p.ChannelMetadata,
 		MessagesCount:    0,
 		AttachmentsCount: 0,
-		StartTime:        0,
-		EndTime:          0,
+		StartTime:        p.BatchStartTime,
+		EndTime:          p.BatchEndTime,
```

Comment:
Is this a functional change?

Reply:
Yes, it will bring global relay's export in line with Actiance and CSV. It will be in the blog post. (And I'm currently refactoring all the tests so that the same ones can be used for all three export types, that's why there aren't a ton of additional tests in this PR).